content-security-policy/base-uri/base-uri_iframe_sandbox.sub.html

```diff
@@ -0,0 +1,76 @@
+<!DOCTYPE HTML>
+<html>
+
+<head>
+    <meta charset="utf-8">
+    <title>base-uri works correctly inside a sandboxed iframe.</title>
+    <script src='/resources/testharness.js'></script>
+    <script src='/resources/testharnessreport.js'></script>
+
+    <!-- CSP served: base-uri 'self' -->
+</head>
+
+<body>
+    <h1>base-uri works correctly inside a sandboxed iframe.</h1>
+    <div id='log'></div>
+
+    <script>
+        window.addEventListener('securitypolicyviolation', function(e) {
+            assert_unreached('No CSP violation report has fired.');
+        });
+
+        async_test(function(t) {
+            window.addEventListener('message', t.step_func(function(e) {
+                if (e.data.startsWith('SAME-ORIGIN,')) {
+                    assert_equals(e.data, 'SAME-ORIGIN,' + location.origin + '/base/');
+                    t.done();
+                }
+            }));
+            var e = document.createElement('iframe');
+            e.sandbox = 'allow-scripts';
+            e.style.display = 'none';
+            e.srcdoc = `
+            <script>
+              window.addEventListener('securitypolicyviolation', function() {
+                top.postMessage('SAME-ORIGIN,FAIL', '*');
+              });
+            </sc` + `ript>
+            <base href="{{location[scheme]}}://{{domains[]}}:{{ports[http][0]}}/base/">
+            <script>
+              top.postMessage('SAME-ORIGIN,' + document.baseURI, '*');
```

Comment:
I'm not sure about the timing here; I think there will end up being a race between the parser picking up the `<script>` tag and blocking execution until the end of the task, and the `securitypolicyviolation` event being fired.

You could mitigate that risk by wrapping this in something like `requestAnimationFrame(_ => top.postMessage(...))`.